### ISSUE TITLE
Markdown preview in browser window

### ISSUE THREAD

**Issue by krstp** [zed-industries/zed#4791]
### Check for existing issues

- [X] Completed

### Describe the feature

I would like to have CTRL+SHIFT+P option to view created Markdown. Upon execution, ZED should parse/compile the Markdown source and display in browser of choice window.

### If applicable, add mockups / screenshots to help present your vision of the feature

![Markdown](https://user-images.githubusercontent.com/4711498/232623171-6be7be8d-1f8a-4fb4-bbd3-4ba25ad62139.png)
extra feature could allow for parsing/compiling of markdown to given platform supporter formatting such as: Vanilla Markdown, GitHub Markdown, BB Markdown etc.
![select](https://user-images.githubusercontent.com/4711498/232623265-a741d723-0b9b-46e2-9fdb-4d40a51d651c.png)

**Comment by jansol** [comment]
Any particular reason why specifically compiling to HTML and opening in a web browser? As opposed to a builtin preview renderer as mentioned in #5064 and #5346

**Comment by krstp** [comment]
Indeed, would be good to see average repo format represented, either it is GitHub/GitLab/BB etc. all of those generally have similar format, or for example what shows up in the Jupyter Notebook rendering etc. Currently, as much as ZED has kick-ass preview feature. it is somewhat custom zed-ish preview formatting, somewhat shows how Markdown looks like, but does not have exactly familiar formatting. Whenever I prepare the Markdown, despite using correct block tags, I would like to see how most of the general populous will see it. I hope this explain. Thanks.

**Comment by JosephTLyons** [comment]
Our built-in Markdown rendered is starting to round out nicely - is there still a desire to have a feature to open the rendered view within a browser, @krstp?

**Comment by jansol** [comment]
Compiling to actual HTML with $(static site generator) seems better suited for a task at this point (can we assign tasks to be run when specific files are saved?)

**Comment by JosephTLyons** [comment]
I dont think we have any hook system built in yet, but that would be a great addition to tasks.

**Comment by notpeter** [comment]
Zed does not current ship a Markdown->HTML renderer.
As a workaround this is relatively trivial to implement with tasks.
Here's an example of how to that uses the GitHub API, Curl, JQ and `open` to spawn the browser:

```json
[
  {
    "label": "markdown preview",
    "command": "jq --slurp --raw-input '{\"text\": \"\\(.)\", \"mode\": \"markdown\"}' < ${ZED_FILENAME} | curl --data @- https://api.github.com/markdown > /tmp/markdown-preview.html && open /tmp/markdown-preview.html"
  }
]
```

Assigning the task a [custom keyboard shortcut](https://zed.dev/docs/tasks#custom-keybindings-for-tasks) is left as exercise to the reader.
